pyk/src/pyk/subst.py

```diff
@@ -46,6 +55,51 @@ def replace(term):
 
         return bottom_up(replace, term)
 
+    def unapply(self, term: KInner) -> KInner:
+        new_term = term
+        for var_name in self:
+            new_term = replaceAnywhereWith((self[var_name], KVariable(var_name)), new_term)
+        return new_term
```

Comment:
We need a way to take into account that not all substitutions should be reversed on the input term.

For example `CALLVALUE_CELL ==Int 0` should not be reversed, because then a bunch of other `0` in the configuration, which are not related to `CALLVALUE_CELL`, get reversed.

For example, here: https://github.com/runtimeverification/erc20-verification/blob/master/ksummarize/src/ksummarize/util.py#L111 we filter to only consider unapplying substitutions where the value being replaced with a variable is not a KToken.

Reply:
I think this can be achieved by masking the subterms that should not be reveresed by fresh variables before unapplying. Would that work? If so, we can add a helper function for that.

Reply:
I don't understand this, what you're doing by filtering for `KToken` is already enough.